src/utils/include/utils/math/orthonormal_vec.hpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2010-2019 The ESPResSo project
+ *
+ * This file is part of ESPResSo.
+ *
+ * ESPResSo is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * ESPResSo is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef ESPRESSO_ORTHONORMAL_VEC_HPP
+#define ESPRESSO_ORTHONORMAL_VEC_HPP
+
+#include "utils/Vector.hpp"
+#include "utils/constants.hpp"
+
+namespace Utils {
+/**
+ * @brief Return a vector that is orthonormal to vec
+ */
+template <typename T, std::size_t N>
+Vector<T, N> calc_orthonormal_vector(Vector<T, N> const &vec) {
```

Comment:
i think it's sufficient to write a version for Vector<T,3>. I'm also not sure if your version would work for an arbitrary dimensional vector.

Reply:
it will
But if 3 is enough for espresso, so be it